brayns/Brayns.h

```diff
@@ -152,7 +155,31 @@ class Brayns
      */
     BRAYNS_API AbstractManipulator& getCameraManipulator();
 
+    /**
+     * Create and add the given plugin type to Brayns. The constructor signature
+     * of the plugin must be:
+     * @code
+     * MyPlugin(EnginePtr, ParametersManager&, ActionInterface*)
+     * @endcode
+     *
+     * @note the ActionInterface can be nullptr, so handle this in your plugin
+     *       accordingly.
+     */
+    template <typename T>
+    std::shared_ptr<T> addPlugin()
+    {
+        auto plugin{std::make_shared<T>(_engine, _parametersManager,
+                                        _actionInterface.get())};
+        _extensionPluginFactory.add(plugin);
+        return plugin;
+    }
+
 private:
+    EnginePtr _engine;
+    ParametersManager _parametersManager;
+    ExtensionPluginFactory _extensionPluginFactory;
+    std::shared_ptr<ActionInterface> _actionInterface;
```

Comment:
When there is a pimpl, in principle all the private members (except virtual functions) should go in it

Reply:
Agreed in principle. This is hopefully just needed for now until we have runtime plugin loading, so this can be private again.